### ISSUE TITLE
Issues with Splash Attention

### ISSUE THREAD

**Issue by versae** [stanford-crfm/levanter#589]
I've been testing both the Splash Attention and the good 'ol vanilla Flash Attention implementations. And it seems that the learning rate affects them very differently. In fact, in my experiments, Splash Attention always seems to underperform and it's harder to get right.

For example, using learning rates 1e-4, 3e-4, and 1e-5, the vanilla implementation (blue-ish) always converges but Splash Attention (red-ish) never gets quite as good.

<img width="636" alt="image" src="https://github.com/stanford-crfm/levanter/assets/173537/c0d7da78-8bd5-4a69-84cd-d95df8d9c7a0">

I also tried with a different dataset, in this case books instead of newspapers. Same.

<img width="636" alt="image" src="https://github.com/stanford-crfm/levanter/assets/173537/f6fff01f-70c5-4421-84ab-b760beadb2e7">

Splash Attention only seems to shine in tokens per second, where it gets around 4000 more tokens per second than the vanilla implementation.

<img width="636" alt="image" src="https://github.com/stanford-crfm/levanter/assets/173537/57de029f-6690-4dc1-a85b-8428c997d2c5">

At this point, I think it would be useful to have a way to select which implementation you want to run.
Also, is there a way to avoid doing evals at the beginning of the training?

**Comment by dlwh** [comment]
Ugh yeah I'll disable/turn into opt in until we figure this out. Thanks for the report. I should have tested it more carefully. The loss curve was approximately the same in my quick test but apparently it's no good.

**Comment by dlwh** [comment]
i bet i messed up mqa

**Comment by dlwh** [comment]
pretty sure this will be fixed in #596 (though I'll make it default on after i test it a bit more)

**Comment by dlwh** [comment]
@versae can you try latest main with:

```
--model.attn_backend splash
```

and  (the default:)

```
--model.attn_backend jax_flash
```

and lmk if things seem good? pretty sure it's correct now

**Comment by versae** [comment]
Nice catch BTW

https://github.com/stanford-crfm/levanter/pull/596/files#diff-fbe2e2b3c420db2356f984aa3ecff7877ca98a9391bd8c9d2139bb9f3c61e100R763

**Comment by dlwh** [comment]
thanks. I should have caught it before, but glad it's fixed now

**Comment by dlwh** [comment]
declaring it fixed in #598
